browser/ui/views/location_bar/brave_location_bar_view.cc

```diff
@@ -279,6 +292,32 @@ std::vector<views::View*> BraveLocationBarView::GetTrailingViews() {
   return views;
 }
 
+void BraveLocationBarView::RefreshBackground() {
+  LocationBarView::RefreshBackground();
+
+  const auto* const color_provider = GetColorProvider();
+  if (!color_provider) {
+    return;
+  }
+
+  const bool is_caret_visible = omnibox_view_->model()->is_caret_visible();
+  if (IsMouseHovered() && !is_caret_visible) {
+    const int radius = GetBorderRadius();
+    if (!shadow_) {
+      ViewShadow::ShadowParameters shadow{
+          .offset_x = 0,
+          .offset_y = 1,
+          .blur_radius = radius,
+          .shadow_color =
+              color_provider->GetColor(kColorLocationBarHoveredShadow)};
+
+      shadow_ = std::make_unique<ViewShadow>(this, radius, shadow);
+    }
+  } else {
+    shadow_.reset();
```

Comment:
Alternatively, we could add a method to `ViewShadow` to get/set the shadow visibility. I don't have enough expertise with the layers API to know if that would be better or not.

Reply:
Yeah, it would be great if we can control for existing shadow instead of reset.

Reply:
Feel free to add something like `SetVisible` to the class definition if you like.

Reply:
Done! - https://github.com/brave/brave-core/pull/20535/commits/bff86324b077ae790532c3a8196675b7b8d5fac6